server/sonar-db-dao/src/main/resources/org/sonar/db/purge/PurgeMapper.xml

```diff
@@ -185,10 +181,8 @@
   <delete id="deleteComponents" parameterType="map">
     delete from projects
     where
-      uuid in
-      <foreach collection="componentUuids" open="(" close=")" item="componentUuid" separator=",">
-        #{componentUuid,jdbcType=VARCHAR}
-      </foreach>
+      project_uuid = #{rootUuid,jdbcType=VARCHAR}
+      or uuid = #{rootUuid,jdbcType=VARCHAR}
```

Comment:
"uuid" is supposed to equal "project_uuid" on root components, but ok, this can be a safeguard.